processing/src/main/java/io/druid/granularity/QueryGranularity.java

```diff
@@ -108,4 +110,22 @@ else if(o instanceof Number)
     }
     throw new IAE("Cannot convert [%s] to QueryGranularity", o.getClass());
   }
+
+  //simple merge strategy on query granularity that checks if all are equal or else
+  //returns null. this can be improved in future but is good enough for most use-cases.
+  public static QueryGranularity mergeQueryGranularities(List<QueryGranularity> toMerge)
+  {
+    if (toMerge == null || toMerge.size() == 0) {
+      return null;
+    }
+
+    QueryGranularity result = toMerge.get(0);
+    for (int i = 1; i < toMerge.size(); i++) {
+      if (!Objects.equals(result, toMerge.get(i))) {
+        return null;
```

Comment:
would it make sense to return the coarser granularity in case of mismatch ? i.e the minimum granularity for the interval that the user can query. e.g If there are segments for both MINUTE and HOUR granularity, return HOUR instead of null.